circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -18,14 +20,14 @@ trait CirceInstances {
       )
     }
 
-  implicit val jsonEncoder: EntityEncoder[Json] =
-    EntityEncoder[String].contramap[Json] { json =>
-      // Comment from ArgonautInstances (which this code is based on):
-      // TODO naive implementation materializes to a String.
-      // See https://github.com/non/jawn/issues/6#issuecomment-65018736
-      Printer.noSpaces.pretty(json)
-    }.withContentType(`Content-Type`(MediaType.`application/json`))
+  implicit def jsonEncoder: EntityEncoder[Json] =
```

Comment:
That was my first version. The problem with that is that you end up having to specify that implicit guy in many places and get very repetitive and easy to forgot some place, which means you would end up using the default `printer` without noticing. With the `abstract printer`, you have to define your on printer only once and then import your customized `CirceInstances` instead of the library one. This way, using the default vs. the custom printer printer is more explicit and thus easier to debug when things go bad.